src/instana/instrumentation/kafka/confluent_kafka_python.py

```diff
@@ -0,0 +1,165 @@
+
+        with tracer.start_as_current_span(
+            "kafka-producer", span_context=parent_context, kind=SpanKind.PRODUCER
+        ) as span:
+            span.set_attribute("kafka.service", args[0])
+            span.set_attribute("kafka.access", "produce")
+
+            # context propagation
+            headers = args[6] if len(args) > 6 else kwargs.get("headers", {})
+            tracer.inject(
+                span.context,
+                Format.KAFKA_HEADERS,
+                headers,
+                disable_w3c_trace_context=True,
+            )
+
+            try:
+                res = wrapped(*args, **kwargs)
+            except Exception as exc:
+                span.record_exception(exc)
+            else:
+                return res
+
+    def trace_kafka_consume(
+        wrapped: Callable[..., InstanaConfluentKafkaConsumer.consume],
+        instance: InstanaConfluentKafkaConsumer,
+        args: Tuple[int, str, Tuple[Any, ...]],
+        kwargs: Dict[str, Any],
+    ) -> List[confluent_kafka.Message]:
+        if tracing_is_off():
+            return wrapped(*args, **kwargs)
+
+        tracer, parent_span, _ = get_tracer_tuple()
+
+        parent_context = (
+            parent_span.get_span_context()
+            if parent_span
+            else tracer.extract(
+                Format.KAFKA_HEADERS, {}, disable_w3c_trace_context=True
+            )
+        )
+
+        with tracer.start_as_current_span(
+            "kafka-consumer", span_context=parent_context, kind=SpanKind.CONSUMER
+        ) as span:
+            span.set_attribute("kafka.access", "consume")
+
+            try:
+                res = wrapped(*args, **kwargs)
+                span.set_attribute(
+                    "kafka.service",
+                    res[0].topic() if isinstance(res, list) and len(res) > 0 else None,
+                )
+            except Exception as exc:
+                span.record_exception(exc)
+            else:
+                return res
+
+    def trace_kafka_poll(
+        wrapped: Callable[..., InstanaConfluentKafkaConsumer.poll],
+        instance: InstanaConfluentKafkaConsumer,
+        args: Tuple[int, str, Tuple[Any, ...]],
+        kwargs: Dict[str, Any],
+    ) -> Optional[confluent_kafka.Message]:
+        if tracing_is_off():
+            return wrapped(*args, **kwargs)
+
+        tracer, parent_span, _ = get_tracer_tuple()
+
+        parent_context = (
+            parent_span.get_span_context()
+            if parent_span
+            else tracer.extract(
+                Format.KAFKA_HEADERS, {}, disable_w3c_trace_context=True
+            )
+        )
+
+        with tracer.start_as_current_span(
+            "kafka-consumer", span_context=parent_context, kind=SpanKind.CONSUMER
+        ) as span:
+            span.set_attribute("kafka.access", "poll")
+
+            try:
+                res = wrapped(*args, **kwargs)
+                span.set_attribute("kafka.service", res.topic() if res else None)
```

Comment:
Suggesting this change because `kafka.service` will be set to `None` [here](https://github.com/instana/python-sensor/pull/700/files#diff-93ccde7c403c9606145fbbc078ce3338b1f052099504b9c0172bae66564ade5bL373) even if we don't set the value below.
```suggestion
                if res:
                  span.set_attribute("kafka.service", res.topic())
```

Reply:
Good catch @GSVarsha. Thanks.